### PR TITLE
feat: add useRouter property to sidenav

### PR DESCRIPTION
**Pull request by klaascuvelier**
Closes carbon-design-system/carbon-components-angular#2811

This MR adds a `useRouter` option to the SideNav component (which is `false` by default). 
When the `input` is `true`, `routerLink` will be used instead.

#### Changelog

**New**

* adds `useRouter` input to `SideNav`

### CODE REVIEW

**Comment by klaascuvelier** [comment]
recheck

**Comment by klaascuvelier** [comment]
recheck

**Comment by klaascuvelier** [comment]
recheck

**Review by Akshat55 — CHANGES_REQUESTED** [review]
Apologies for the late review, I've been juggling multiple projects this month 😓 Overall PR is good to go, just minor linting changes.

Im targeting a release for this PR early next week.

**Comment by klaascuvelier** [comment]
Thanks for the review @Akshat55!  Took care of your remarks. 

Quick question, did you do manual linting here? If so, is there a possibility to automate some things? Like adding eslint or prettier to the repo maybe?

**Comment by Akshat55** [comment]
> Quick question, did you do manual linting here? If so, is there a possibility to automate some things? Like adding eslint or prettier to the repo maybe?

We do some manual linting here and there while reviewing the code. We do have a linter (tslint) and we run it prior to pushing the commit (using husky). 

We haven't updated to eslint in v5 because we noticed it was taking way too long. From what I've seen, the duration has decreased and we plan to replace our existing linter with eslint in v6!

File: src/ui-shell/sidenav/sidenav-item.component.ts [hunk 1]
```diff
@@ -16,22 +17,37 @@ import { Router } from "@angular/router";
...
+			<ng-template [ngTemplateOutlet]="sidenavItemContentTpl"></ng-template>
+		</a>
+
+		<ng-template #sidenavItemRouterTpl>
+			<a
+				[routerLink]="route"
+				[routerLinkActive]="'cds--side-nav__item--active'"
+				[ariaCurrentWhenActive]="'page'"
+				[attr.title]="title ? title : null"
+				class="cds--side-nav__link">
```
Comment (Akshat55) [inline src/ui-shell/sidenav/sidenav-item.component.ts:38]: I don't think we need to bind routerlinkactive & ariaCurrentWhenActive here. We should be able to just do:

```suggestion
				[routerLink]="route"
				routerLinkActive="cds--side-nav__item--active"
				ariaCurrentWhenActive="page"
				[attr.title]="title ? title : null"
				class="cds--side-nav__link">
```

File: src/ui-shell/ui-shell.stories.ts [hunk 2]
```diff
@@ -303,6 +309,51 @@ const SideNavigationRouterTemplate = (args) => ({
...
+			    position: absolute;
+    			background-color: var(--cds-border-interactive, #0f62fe);
+    			content: "";
+    			inline-size: 3px;
+    			inset-block-end: 0;
+    			inset-block-start: 0;
+    			inset-inline-start: 0;
+    		}
+		}
+	`],
```
Comment (Akshat55) [inline src/ui-shell/ui-shell.stories.ts:329]: I don't think we need this. After looking at @carbon/styles, seems like the `active` class is assigned when `aria-current="page"`. Could you please verify this when router is enabled?

**Review by Akshat55 — CHANGES_REQUESTED** [review]

File: src/ui-shell/sidenav/sidenav-item.component.ts (hunk 1 above)
Comment (klaascuvelier) [reply src/ui-shell/sidenav/sidenav-item.component.ts:38]: You're absolutely right

File: src/ui-shell/ui-shell.stories.ts (hunk 2 above)
Comment (klaascuvelier) [reply src/ui-shell/ui-shell.stories.ts:329]: You are also correct, this was probably an issue before I correctly added the `aria-current` thanks for noticing!

**Comment by klaascuvelier** [comment]
> > Quick question, did you do manual linting here? If so, is there a possibility to automate some things? Like adding eslint or prettier to the repo maybe?
> 
> We do some manual linting here and there while reviewing the code. We do have a linter (tslint) and we run it prior to pushing the commit (using husky).
> 
> We haven't updated to eslint in v5 because we noticed it was taking way too long. From what I've seen, the duration has decreased and we plan to replace our existing linter with eslint in v6!

Ah right, noticed a warning in my console when pushing/committing. Prolly I should look into resolving that.

**Review by Akshat55 — APPROVED** [review]

**Comment by Licen-it** [comment]
This PR enabled the routing by default, but didn't take in consideration the `routeExtras` passed to the component (which is breaking anyone using the `relativeTo` option)

**Pull request merged**